clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/cache/AzureLoadBalancerCachingAgent.groovy

```diff
@@ -75,30 +91,223 @@ class AzureLoadBalancerCachingAgent implements CachingAgent, AccountAware {
   }
 
   @Override
+  String getOnDemandAgentType() {
+    "${getAgentType()}-OnDemand"
+  }
+
+  @Override
   Collection<AgentDataType> getProvidedDataTypes() {
     return types
   }
 
   @Override
+  boolean handles(String type) {
+    type == LEGACY_ON_DEMAND_TYPE
+  }
+
+  @Override
+  boolean handles(String type, String cloudProvider) {
+    type == ON_DEMAND_TYPE && cloudProvider == azureCloudProvider.id
+  }
+
+  @Override
+  OnDemandAgent.OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    if (!data.containsKey("loadBalancerName") ||
+        !data.containsKey("account") ||
+        !data.containsKey("region")  ||
+        accountName != data.account ||
+        region != data.region) {
+      return null
+    }
+
+    AzureLoadBalancerDescription updatedLoadBalancer = null
+    AzureLoadBalancerDescription evictedLoadBalancer = null
+    String loadBalancerName = data.loadBalancerName as String
+    String resourceGroupName = AzureUtilities.getResourceGroupName(AzureUtilities.getAppNameFromAzureResourceName(loadBalancerName), region)
+    if (resourceGroupName == null) {
+      log.info("handle->Unexpected error retrieving resource group name: null")
+      return []
+    }
+
+    try {
+      updatedLoadBalancer = metricsSupport.readData {
+        creds.networkClient.getLoadBalancer(creds, resourceGroupName, loadBalancerName)
+      }
+    } catch (Exception e ) {
+      log.info("handle->Unexpected exception: ${e.message}")
+      return null
```

Comment:
Silently failing on an onDemand request is probably a little less than idea. 

It could cause subsequent steps to proceed with potentially out-of-date information.